source/extensions/health_checkers/http/health_checker_impl.cc

```diff
@@ -81,15 +81,25 @@ HttpHealthCheckerImpl::HttpHealthCheckerImpl(const Cluster& cluster,
     service_name_matcher_.emplace(config.http_health_check().service_name_matcher());
   }
 
+  if (!Http::HeaderUtility::authorityIsValid(host_value_)) {
```

Comment:
@yanavlasov how do HCs fit with UHV?

Reply:
We have discussed with @botengyao . Presently there is no way to use UHV since it is still behind `#ifdef`, so just adding PGV validator to avoid ASSERTs in the fuzzer should be sufficient.

@botengyao please add a TODO here to introduce additional validation for the authority and path headers based on the default UHV when it is available.

Reply:
Done, thanks!

Reply:
I think you can just PGV annotations to the config proto and avoid these custom checks here altogether.

Reply:
Thanks for the review!

Per discussion with @adisuissa  we want some more clarity on whether a stricter host and path validations are needed in this case, decided to use this customized logic. 

Here is the PGV set which is only header value check: https://github.com/bufbuild/protoc-gen-validate/blob/7d833c1f32da0f46ebecc1f3f88cf093cd73a6e6/python/protoc_gen_validate/validator.py#L27

There are some small difference for path and header:

```
constexpr absl::string_view kValidAuthorityChars =
    "abcdefghijklmnopqrstuvwxyzABCDEFGHIJKLMNOPQRSTUVWXYZ0123456789-._~%!$&'()["
    "]*+,;=:";

constexpr absl::string_view kValidPathChars =
    "/abcdefghijklmnopqrstuvwxyzABCDEFGHIJKLMNOPQRSTUVWXYZ0123456789-._~%!$&'()"
    "*+,;=:@?";
```

I am fine to directly use the HTTP_HEADER_VALUE PGV check, but current check is stricter.

Reply:
Discussed offline, and will use PGV right now till UHV is available.